### PR TITLE
Feat: Two-Factor Authentication - (enable)

### PR DESCRIPTION
**Pull request by RB-Isiaq**
## What does this PR do?

This PR is about enabling two factor authentication for additional layer of security.

## How should this be manually tested?

As an authenticated user, send a POST request to `/api/v1/auth/2fa/enable` with a request body as below:

  ```json
  {
    "password": "string",
  }
```
## Checklist of what you did

- [x] Implemented endpoint to enable 2fa authentication.
- [x] Used a secure TOTP algorithm compatible with common authenticator apps (speakeasy library).
- [x] Generate and store a unique secret key for each user enabling 2FA.
- [x] Provide a QR code for easy secret key input into authenticator apps.
- [x] Returned success message with the right response payload.
- [x] Added unit tests for the auth service

## Success Response 
Screenshot 2024-07-24 101542
Screenshot 2024-07-24 100627

## Error Response 
Screenshot 2024-07-24 101943
Screenshot 2024-07-24 101646
Screenshot 2024-07-24 101624

## Linked Issue
#266

### CODE REVIEW

File: db/migrations/1721758146468-migration.ts
Comment (Homoakin619) [inline db/migrations/1721758146468-migration.ts:1]: Migration file is empty, kindly remove

File: src/modules/user/user.service.ts [hunk 1]
```diff
@@ -20,6 +21,10 @@ export default class UserService {
     await this.userRepository.save(newUser);
   }
 
+  async createUserSecret(user: UserResponseDTO) {
+    await this.userRepository.save(user);
```
Comment (Homoakin619) [inline src/modules/user/user.service.ts:25]: This should be an update action you have here

**Comment by RB-Isiaq** [comment]
Thank you @Homoakin619 
Conversations have been resolved.

**Review by Homoakin619 — APPROVED** [review]

**Comment by RB-Isiaq** [comment]
@buka4rill Please review. Thanks.

**Review by Cobidex — APPROVED** [review]

**Review by nedssoft — COMMENTED** [review]
This task seems incomplete, there's no implementation for infusing the 2FA as a part of the authentication flow. 

If integrating it to the auth flow is beyond the scope of the task, then it looks good to me

**Comment by RB-Isiaq** [comment]
> This task seems incomplete, there's no implementation for infusing the 2FA as a part of the authentication flow.
> 
> If integrating it to the auth flow is beyond the scope of the task, then it looks good to me

@nedssoft Thank you for the feedback.

The 2FA flow was split into sub tasks by @Heba-WebDev 

I handled the enabling flow, Some other persons handle the verification flow, deactivation flow, and backup flow.

Later on in a follow up pr, the verification could be handled in the login endpoint for users that have 2fa enabled.

So, Integrating it into the auth flow is beyond this task, as this task is only scoped for enabling the 2fa authentication.

**Pull request closed without merging**